supersim_test.go

```diff
@@ -146,3 +168,53 @@ func TestAccountBalances(t *testing.T) {
 		}
 	}
 }
+
+func TestDepositTxSimpleEthDeposit(t *testing.T) {
+	testSuite := createTestSuite(t)
+
+	l1Anvil := testSuite.Supersim.Orchestrator.L1Anvil()
+	l1EthClient, _ := ethclient.Dial(l1Anvil.Endpoint())
+
+	var wg sync.WaitGroup
+	var l1TxMutex sync.Mutex
+
+	wg.Add(len(testSuite.Supersim.Orchestrator.OpSimInstances))
+	for i, opSim := range testSuite.Supersim.Orchestrator.OpSimInstances {
+		go func() {
+			defer wg.Done()
+
+			l2EthClient, _ := ethclient.Dial(opSim.Endpoint())
+			privateKey, _ := testSuite.HdAccountStore.DerivePrivateKeyAt(uint32(i))
+			senderAddressHex, _ := testSuite.HdAccountStore.AddressHexAt(uint32(i))
+			senderAddress := common.HexToAddress(senderAddressHex)
+
+			oneEth := big.NewInt(1e18)
+			prevBalance, _ := l2EthClient.BalanceAt(context.Background(), senderAddress, nil)
+
+			transactor, _ := bind.NewKeyedTransactorWithChainID(privateKey, big.NewInt(int64(l1Anvil.ChainID())))
+			transactor.Value = oneEth
+			optimismPortal, _ := bindings.NewOptimismPortal(opSim.L2Config.L1DeploymentAddresses.OptimismPortalProxy, l1EthClient)
+
+			// needs a lock because the gas estimation can become outdated between transactions
+			l1TxMutex.Lock()
+			tx, err := optimismPortal.DepositTransaction(transactor, senderAddress, oneEth, 100000, false, make([]byte, 0))
+			l1TxMutex.Unlock()
+			require.NoError(t, err)
+
+			txReceipt, _ := bind.WaitMined(context.Background(), l1EthClient, tx)
+			require.NoError(t, err)
+
+			require.True(t, txReceipt.Status == 1, "Deposit transaction failed")
+			require.NotEmpty(t, txReceipt.Logs, "Deposit transaction failed")
+
+			// wait for the deposit to be processed
+			time.Sleep(1 * time.Second)
```

Comment:
is there something else we can wait on to avoid adding sleep to the test?

Reply:
https://github.com/ethereum-optimism/optimism/blob/develop/op-e2e/e2eutils/wait/waits.go#L35 

`wait.ForReceiptOk` is a good usecase for this. Can add a context that times out after 1 minute

Reply:
I looked into this, and I think the deposit tx hash generated from the logs is inconsistent with the resulting L2 transaction hash. Might be an issue with how i submit deposit tx, might be an issue with anvil for deposit tx types.

For now, I'll keep it to just waiting, and once I resolve the issue https://github.com/ethereum-optimism/supersim/issues/64 will update the sleep timer